cadc-data-ops-fits/src/main/java/org/opencadc/fits/slice/PolygonCutout.java

```diff
@@ -189,23 +193,19 @@ public PolygonCutout(FITSHeaderWCSKeywords fitsHeaderWCSKeywords) {
         double x2 = -1.0D * x1;
         double y1 = Double.MAX_VALUE;
         double y2 = -1.0D * y1;
+        final int spatialLongitudeAxisIndex = coordSys.longitudeAxis - 1;
+        final int spatialLatitudeAxisIndex = coordSys.latitudeAxis - 1;
+
         LOGGER.debug("Bounding box is " + polygonToCut);
         for (final Point point : polygonToCut.getVertices()) {
-            final double[] spatialCoords = new double[2];
-
-            if (coOrdSys.isSwappedAxes()) {
-                LOGGER.debug("Coordinates are swapped.");
-                spatialCoords[0] = point.getLatitude();
-                spatialCoords[1] = point.getLongitude();
-            } else {
-                spatialCoords[0] = point.getLongitude();
-                spatialCoords[1] = point.getLatitude();
-            }
-
-            // Pad the axes that go beyond what we're interested in (NAXISn where n > 2).
-            final double[] worldCoords = Arrays.copyOf(spatialCoords, naxis);
-
-            for (int i = spatialCoords.length; i < worldCoords.length; i++) {
+            final double[] worldCoords = new double[naxis];
+            // Set the spatial values where they should be.
+            worldCoords[spatialLongitudeAxisIndex] = point.getLongitude();
+            worldCoords[spatialLatitudeAxisIndex] = point.getLatitude();
+
+            // Fill in the rest of the world coordinates.
+            for (int i = 0; i < worldCoords.length
+                            && i != spatialLongitudeAxisIndex && i != spatialLatitudeAxisIndex; i++) {
```

Comment:
Are you sure this loop does what you want?
```
groovy:000> for (int i=0; i<5 && i != 3; i++) { System.out.println(i); }
0
1
2
===> null
```
says it would only fill in `worldCoords` up to but not including `spatialLongitudeAxisIndex` or `spatialLatitudeAxisIndex` but not after.